frontend/src/components/AppMap.vue

```diff
@@ -51,61 +118,71 @@ import {
 import * as d3 from "d3";
 import domtoimage from "dom-to-image";
 import type { GeoJsonProperties } from "geojson";
-import L, { type MapOptions, type TileLayerOptions } from "leaflet";
-import { debounce, round } from "lodash";
-import { useElementSize, useResizeObserver } from "@vueuse/core";
-import type { Geometry } from "@/api";
-import { gradientOptions, type GradientFunc } from "@/components/gradient";
+import L, { type MapOptions } from "leaflet";
+import { debounce, mapValues } from "lodash";
+import {
+  faCropSimple,
+  faDownload,
+  faExpand,
+  faMinus,
+  faPlus,
+} from "@fortawesome/free-solid-svg-icons";
+import { useElementSize, useFullscreen, useResizeObserver } from "@vueuse/core";
+import type { Data, Overlays } from "@/api";
+import AppButton from "@/components/AppButton.vue";
+import { getGradient } from "@/components/gradient";
+import { markerOptions } from "@/components/markers";
+import { useScrollable } from "@/util/composables";
 import { downloadPng } from "@/util/download";
+import { formatValue, normalizedApply } from "@/util/math";
+import { getBbox, sleep } from "@/util/misc";
 import "leaflet/dist/leaflet.css";
 
-// ref to root element
+/** element refs */
+const scroll = ref<HTMLDivElement>();
 const element = ref<HTMLDivElement>();
 
 type Props = {
-  // map pan/zoom
-  lat?: number;
-  long?: number;
-  zoom?: number;
-  // geojson
-  geometry?: Geometry;
-  // show/hide legend
-  showLegend?: boolean;
-  // value domain
+  /** data */
+  data?: Data;
+  overlays?: Overlays;
+  /** map of feature id to value */
+  values?: { [key: number]: number };
+  /** value domain */
   min?: number;
   max?: number;
-  // map of feature id to value
-  values?: { [key: number]: number };
-  // number of scale steps
-  steps?: number;
-  // color gradient id
-  gradient?: string;
-  // reverse direction of gradient
-  flipGradient?: boolean;
-  // forced dimensions
-  width?: number;
-  height?: number;
-  // layer opacities
-  baseOpacity?: number;
-  dataOpacity?: number;
+  /** map pan/zoom */
+  lat: number;
+  long: number;
+  zoom: number;
+  /** show/hide elements */
+  showLegends: boolean;
+  /** layer opacities */
+  baseOpacity: number;
+  dataOpacity: number;
+  overlayOpacity: number;
+  /** tile providers for layers */
+  base: string;
+  /** color gradient id */
+  gradient: string;
+  /** scale props */
+  flipGradient: boolean;
+  scaleSteps: number;
+  niceSteps: boolean;
+  scalePower: number;
+  /** forced dimensions */
+  width: number;
+  height: number;
+  /** filename for download */
+  filename: string | string[];
 };
 
 const props = withDefaults(defineProps<Props>(), {
-  lat: undefined,
-  long: undefined,
-  zoom: undefined,
-  geometry: () => [],
-  showLegend: true,
-  min: 0,
-  max: 0,
+  data: () => ({ type: "FeatureCollection", features: [] }),
+  overlays: () => ({}),
   values: () => ({}),
-  steps: 5,
-  gradient: "interpolateCool",
-  flipGradient: false,
```

Comment:
Remove defaults here and just require these properties, because they're always being passed from parent. Default just becomes default value set in parent variable ref.